src/utils/styleProps.ts

```diff
@@ -156,11 +161,14 @@ function getPropNamesFromPropDefinition(propDefinition) {
  * ```
  */
 
-function getPropFilter(propNames) {
+function getPropFilter(propNames: string[]): (prop: string | number) => boolean  {
```

Comment:
Prop names are always strings:

```suggestion
function getPropFilter(propNames: string[]): (prop: string) => boolean  {
```

I'm not sure why I'm doing the `typeof` check further down, maybe overly defensive coding... that can probably be cleaned up.

Reply:
The condition is still needed but i've rewritten it slightly different now.

This is because `propsToFilter` is a js Set, and only strings are allowed to be passed into it's `has()` method. So even though this function accepts string | number we need to case it to a string at some point.

this is the nice stuff TS catches for us.

Reply:
I'm just confused about why the returned function needs to accept a param of type `number`, as JSX prop names can only be strings.